File: Coding/Java/Set2/FirstSmallestSecondSmallest.java
package Set2;

public class FirstSmallestSecondSmallest {

    public static int[] firstNSecondSmallest(int[] arr){
        int smallest = Integer.MAX_VALUE;
        int second_smallest = Integer.MAX_VALUE;

        for(int i= 0;i<arr.length;i++){
            if(arr[i] < smallest){
                second_smallest = smallest;
                smallest = arr[i];
            } else if(arr[i] < second_smallest && arr[i] != smallest){
                second_smallest = arr[i];
            }
        }

        return new int[]{smallest, second_smallest};
    }

    public static void main(String[] args) {
        int[] arr = {1,3,4,6,4,2};

        int[] val = firstNSecondSmallest(arr);
        System.out.println("smallest is " + val[0]);
        if(val[1]!= Integer.MAX_VALUE){
            System.out.println("second smallest is " + val[1]);
        } else {
            System.out.println("second smallest doesn't exist");
        }
    }
}

File: Coding/JavaScript/Set2/FirstSmallestSecondSmallest.js
let arr = [1,3,4,6,4,2]

val = first_small_second_small(arr)
console.log(`smallest value is ${val[0]}`)
if(val[1] == Number.MAX_VALUE)
    console.log(`second smallest value is not present`)
else
    console.log(`second smallest value is ${val[1]}`)

function first_small_second_small(arr){
    let smallest = Number.MAX_VALUE
    let second_smallest = Number.MAX_VALUE

    for(val of arr){
        if(val < smallest){
            second_smallest = smallest
            smallest = val
        } else if(val < second_smallest && val != smallest){
            second_smallest = val
        }
    }

    return [smallest, second_smallest]
}

